app/samples/openrocket_sensors/src/main.cpp

```diff
@@ -0,0 +1,39 @@
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+
+#include <f_core/device/sensor/c_accelerometer.h>
+#include <f_core/device/sensor/c_barometer.h>
+#include <f_core/device/sensor/c_gyroscope.h>
+
+int main() {
+    CAccelerometer imu_accelerometer(*DEVICE_DT_GET_ONE(openrocket_imu));
```

Comment:
Would rather keep this in. We should keep the arguments as a reference so we can catch issues at compile time vs at runtime

Reply:
I think they'll have the same runtime/compile time split, right?
If you get a nullptr (from DEVICE_DT_GET_ANY) with the constructing taking a reference youll dereference a null pointer at init time but before your constructor is called vs .when you take a pointer youll get it when your constructor is called during init but within your constructor so you can provide more info.

For the compile time, errors (undefined reference when you forget a driver and use DEVICE_DT_GET or DEVICE_DT_GET_ONE) you will still get a compile time error.

Am i missing something?

Reply:
Youll still get a compile time error if you do a DEVICE_DT_GET_ANY on something that doesnt exist. I tested this. DEVICE_DT_GET* is still evaluated at compile time since it references the generated devicetree header. Passing a nullptr with the constructor taking a pointer will pass compile and just crash on runtime.